lib/io/hardware_i2c.hpp

```diff
@@ -28,10 +28,16 @@ class HardwareI2c : public II2c {
   virtual core::Result writeByteToRegister(const std::uint8_t device_address,
                                            const std::uint8_t register_address,
                                            const std::uint8_t data);
+  virtual core::Result writeByteToRegister(const std::uint8_t device_address,
```

Comment:
as an aside, you were sensors last year do you know why these are virtuals? I don't think we ever inherit (or are likely to?) from this class

Reply:
Don't know, and I agree: I'm not sure when you would ever inherit from HardwareI2C specifically

Reply:
Update: Ishan said there was a reason for this. @TomLonergan03 What was it, and should these virtuals be reinstated as a result?